overlord/snapstate/snapmgr.go

```diff
@@ -229,6 +232,18 @@ func (m *SnapManager) doPrepareSnap(t *state.Task, _ *tomb.Tomb) error {
 		return err
 	}
 
+	// check if we need to read .sideinfo files
+	var si snap.SideInfo
+	if (ss.Flags & WithSideInfo) > 0 {
+		metafn := ss.SnapPath + ".sideinfo"
+		if j, err := ioutil.ReadFile(metafn); err == nil {
```

Comment:
I wonder if this could be done earlier in InstallPathWithSideInfo, change doInstall to take a SnapSetup and have an optional SideInfo in SnapSetup?